### PR TITLE
Extend --live-out CLI with ;nzcv suffix (closes #81)

### PR DESCRIPTION
**Pull request by pmatos**
## Summary

- Adds `validation::live_out::parse_live_out_contract(s) -> Result<(LiveOut, bool), _>` implementing the grammar `<regs>` or `<regs>;<flags>`; group token `nzcv` only, per-flag tokens `n`/`z`/`c`/`v` and bareword `nzcv` (no leading `;`) reserved for a future per-flag rev.
- Routes both `run_equiv` and `run_llm_opt` through the new helper and unifies the previously divergent CLI parse paths under a single `"invalid live-out: ..."` error prefix.
- `run_equiv` plugs the parsed flag bit into `EquivalenceConfig::with_flags(flags_live)` so users can finally opt into NZCV-aware equivalence checking from the CLI (e.g. `--live-out "x0,x1;nzcv"`).
- `run_llm_opt` accepts the suffix for vocabulary parity but discards the bit — `outcome.rs` already pins `flags_live=true` on that verifier path, documented in source + ADR.
- ADR-0006 records the grammar decision and the reservation policy. TUTORIAL.md and README.md document the new suffix.

## Test plan

- [x] `cargo test --bin s11 validation::live_out::tests::test_parse_live_out_contract` — 14 unit tests cover happy paths (`x0,x1;nzcv`, `;nzcv`, `x0,x1;`, `X0,X1;NZCV`, `""`), whitespace edge cases (`x0 ; nzcv`, `;`, `  ;  `), and error classes (bareword `nzcv`, multi-section, unknown flag token, reserved `n`/`z`/`c`/`v`, invalid register).
- [x] `./ci_check.sh` — fmt + build + unit + integration tests pass.
- [ ] Manual sanity (reviewer or local): `s11 equiv a.s b.s --live-out "x0;nzcv" --verbose` prints `Live-out flags: nzcv`; same with `x0,x1` (no suffix) prints only registers; `nzcv` (bareword) errors with `flag-only live-out requires a leading ';'`.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1362,31 +1364,22 @@ fn run_equiv(
...
+            println!("Live-out flags: nzcv");
+        }
     }
 
-    let live_out = LiveOut::from_registers(live_out_regs);
-
-    // Build config
     let config = EquivalenceConfig::default()
         .live_out(live_out)
+        .with_flags(flags_live)
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/main.rs:1382]: Fixed in `9f8e6a4`.

Changed: `run_fast_path` now unions the source registers of both sequences into the random-input mask via a new `fast_path_input_registers` helper. With your `--live-out ';nzcv' --fast-only` example, x1 is now randomized across all 10 random inputs and the 11 edge-case inputs (so `tst x1, #1` vs `tst x1, #2` is correctly caught as `NotEquivalentFast` at the very first non-zero x1). The SMT path is unaffected — it already operates on symbolic inputs.

Validation: new regression test `fast_only_flags_only_contract_detects_tst_divergence` in `src/semantics/equivalence.rs` pins the previously-broken behaviour; the full equivalence suite (51 tests) and `./ci_check.sh` both pass.

_Triaged by Claude Opus 4.7 and Codex GPT-5.4 — both rated FIX (high confidence, soundness)._

**Comment by pmatos** [comment]
@claude Regarding your code review (https://github.com/pmatos/s11/pull/172#issuecomment-4475747275):

Thanks — each of the five sub-items has been triaged by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.4) and processed below.

---

**Item 1 — Error type re-use is slightly misnamed:** **Deferred (diminishing returns)**

Tracked as #180 (https://github.com/pmatos/s11/issues/180).

The evaluators split between defer-as-aesthetic and reject-as-style. To avoid silently dropping the concern, filed a tracking issue. Renaming a public error type has blast radius beyond this PR.

---

**Item 2 — "Unknown" vs "reserved" in error message:** Fixed in `6f1551b`.

Changed: `parse_live_out_contract` now matches `n`/`z`/`c`/`v` as a separate arm with the message `"per-flag token '{}' is reserved for a future extension; use 'nzcv' for all flags"`. Everything else still gets the unknown-token message, now simplified to `"unknown flag token '{}'; expected 'nzcv'"`. The existing `test_parse_live_out_contract_per_flag_tokens_reserved` test was tightened to assert on the new message content (previously only `is_err()`).

Validation: all 14 `parse_live_out_contract` tests pass; `./ci_check.sh` green.

---

**Item 3 — `run_llm_opt` help text doesn't mention the ignored suffix:** Fixed in `675d3ca`.

Changed: `LlmOpt::live_out` help text now reads "Live-out contract (comma-separated regs; ';nzcv' suffix is accepted for syntax compatibility with `equiv` but has no effect here — the LLM verifier always treats NZCV as live; see ADR-0006)".

Validation: build green; `./ci_check.sh` green.

---

**Item 4 — Missing test: reversed order `"nzcv;x0"`:** **Deferred (diminishing returns)**

Tracked as #181 (https://github.com/pmatos/s11/issues/181).

Codex argued that `nzcv;x0` already errors via the existing register parser, so pinning the message adds limited coverage. Evaluators disagreed on FIX vs DEFER; per the autofix rules disagreement defaults to DEFER. The tracking issue captures the rationale for a future cleanup pass.

---

**Item 5 — `semicolon_count` computed on `trimmed` but error displays `s`:** **Style preference (no change)**

Both evaluators rated this REJECT (style-preference, high confidence). The mismatch is intentional — the error string deliberately shows the user's original input (with any whitespace) so they can see exactly what they typed, while internal logic operates on `trimmed`. Adding a comment would be defensive but not necessary; using `trimmed` everywhere would change the diagnostic to show a stripped form. Leaving as-is.

---

_Items 1, 4, and 5 are tracked or rationale-recorded so nothing is silently dropped. Items 2 and 3 are landed in the commits above; CI re-running on the new head._

File: src/semantics/equivalence.rs [hunk 2]
```diff
@@ -235,7 +267,7 @@ fn run_fast_path(
     config: &EquivalenceConfig,
 ) -> Option<EquivalenceResult> {
     let live_out_registers = config.live_out.registers();
-    let input_regs: Vec<_> = live_out_registers.iter().cloned().collect();
+    let input_regs = fast_path_input_registers(live_out_registers, seq1, seq2);
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/semantics/equivalence.rs]: Fixed in `06eaa54`.

Changed: `run_fast_path` now appends 16 extra inputs covering every initial NZCV combination (one per `0..16`, bits assigned to N/Z/C/V) with source registers randomized, **when** `config.flags_live` is true **and** either sequence reads flags before writing them (`reads_flags_before_writing`). Isolated in a new `fast_path_initial_nzcv_variants` helper so register-only sequences pay nothing. Your `ccmp x0,#0,#0,mi` vs `ccmp x1,#1,#0,mi` example now correctly returns `NotEquivalentFast` under `--fast-only --live-out ';nzcv'`: the variant with initial N=true hits the MI branch on both sequences, and the randomized x0/x1 values diverge the result NZCV.

Validation: new regression test `fast_only_flags_only_contract_detects_ccmp_initial_nzcv_divergence` in `src/semantics/equivalence.rs` pins this case; full equivalence suite (52 tests) and `./ci_check.sh` both green.

_Triaged by Claude Opus 4.7 and Codex GPT-5.4 — both rated FIX (high confidence, soundness)._

**Comment by pmatos** [comment]
@claude Regarding your two follow-up reviews (https://github.com/pmatos/s11/pull/172#issuecomment-4476049180 and https://github.com/pmatos/s11/pull/172#issuecomment-4476052187):

**Already handled (no further change)** — both follow-ups are confirmations of items already landed or tracked. Thanks for the verification.

Heads up: while your "ready to merge" verdict was based on `675d3ca`, a second P2 finding from @chatgpt-codex-connector came in immediately afterwards (review thread PRRT_kwDOOuU3Hc6CxjC4, about initial-NZCV variants for flag-reading sequences). That one was triaged FIX and is now landed in commit `06eaa54`. CI re-running on the new head.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.4). If you disagree, please reply and we'll re-evaluate._

File: src/validation/live_out.rs [hunk 3]
```diff
@@ -70,6 +70,60 @@ impl FromStr for LiveOutRegisters {
...
+    if semicolon_count == 0 {
+        if trimmed.eq_ignore_ascii_case("nzcv") {
+            return Err(ParseLiveOutRegistersError {
+                message: format!(
+                    "flag-only live-out requires a leading ';' (e.g. \";nzcv\"); got '{}'",
+                    s
+                ),
+            });
+        }
+        let regs = LiveOutRegisters::from_str(trimmed)?;
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/validation/live_out.rs:109]: Fixed in `81d2157`.

Changed: extended `validation::live_out::parse_register` to recognise `fp` (X29) and `lr` (X30), matching the AArch64 alias set accepted by `src/parser/mod.rs::parse_register`. The pre-PR `equiv` path delegated to that parser directly so it accepted those aliases; the unified `parse_live_out_contract` lost them when it switched to `LiveOutRegisters::from_str`. New tests pin `fp`/`lr` (upper and lowercase) in both `parse_register` and `parse_live_out_contract` (including the `lr;nzcv` form).

Validation: full `validation::live_out` tests (47 total, +4 new) pass; `./ci_check.sh` green.

_Triaged by Claude Opus 4.7 and Codex GPT-5.4 — both rated FIX (high confidence, CLI-contract regression)._

File: src/semantics/equivalence.rs [hunk 4]
```diff
@@ -262,6 +335,26 @@ fn run_fast_path(
...
 
+    // Under `fast_only`, the SMT path that would otherwise catch initial-
+    // NZCV divergence is skipped — for sequences that read flags before
+    // writing them (CSEL family, CCMP, CCMN, ...), also test all 16 initial
+    // NZCV combinations. Gated on `fast_only` because the SMT path already
+    // handles this correctly via symbolic initial state, and adding 16
+    // inputs to every search-algorithm candidate would burn significant
+    // wall-clock time on a verifier that's normally SMT-authoritative.
+    if config.fast_only
+        && config.flags_live
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/semantics/equivalence.rs]: Fixed in `0892fce`.

Changed: dropped `config.flags_live` from the initial-NZCV variants gate in `run_fast_path`. The gate is now `config.fast_only && (reads_flags_before_writing(seq1) || reads_flags_before_writing(seq2))` — so a flag-reading sequence influences `--fast-only` verdicts whenever it reads flags, regardless of whether NZCV is itself live-out. Your `csel x0, x1, x2, mi` vs `mov x0, x2` example under `--live-out x0 --fast-only` is now correctly rejected: the N=true variant hits seq1's MI branch (x0=x1) while seq2 always sets x0=x2, and random x1 != x2 diverges the result.

Validation: new regression test `fast_only_register_contract_detects_csel_initial_nzcv_divergence` pins this case (deliberately uses a register-only contract without `.with_flags(true)`); 53 equivalence tests pass; `./ci_check.sh` green. The 16-input cost remains bounded by the existing `reads_flags_before_writing` gate, and SMT-authoritative callers (every search algorithm) are unaffected — they don't set `fast_only`.

_Triaged by Claude Opus 4.7 and Codex GPT-5.4 — both rated FIX (high confidence, fast-only soundness)._

**Comment by pmatos** [comment]
@claude / @chatgpt-codex-connector — consolidated triage of the latest round of reviews:

---

**Inline P2: alias regression on `src/validation/live_out.rs:101`** (`PRRT_kwDOOuU3Hc6Cxx6a`): **Fixed in `81d2157`**. `fp`/`lr` aliases restored.

**Inline P2: csel-vs-mov register-output gap on `src/semantics/equivalence.rs:345`** (`PRRT_kwDOOuU3Hc6CyJ1I`): **Fixed in `0892fce`**. NZCV-variants gate widened — drops `config.flags_live`, fires on any flag-reading sequence under `--fast-only`.

---

**@claude review (https://github.com/pmatos/s11/pull/172#issuecomment-4476183513 + duplicates 4476283169, 4476336575, plus the earlier ack at 4476167546):**

| Sub-item | Outcome |
|---|---|
| 1. Stale `#[allow(dead_code)]` on `reads_flags_before_writing` | **Fixed in `f1d05ed`** — attribute removed, doc updated to name the actual consumer (`fast_path_initial_nzcv_variants`). |
| 2. PR/thread-ID refs in source comments (CLAUDE.md violation) | **Fixed in `9974275`** — three doc comments stripped of `PR #172`/`PRRT_…` references; technical reasoning preserved. |
| 3. Inline-comment the `count: 16` literal | **Deferred (diminishing returns)** — tracked as #186 (https://github.com/pmatos/s11/issues/186). |
| 4. `"nzcv,x0"` confusing-error follow-up | **Deferred (scope-creep)** — tracked as #187 (https://github.com/pmatos/s11/issues/187). |

---

**@chatgpt-codex-connector follow-up (https://github.com/pmatos/s11/pull/172#issuecomment-4476166396) — "To use Codex here, create an environment for this repo":** **Not applicable (no code change)** — infrastructure note about the bot's review tool setup, not feedback on this PR's contents.

---

_All FIX outcomes are evaluated by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.4) before code change. If you disagree with any verdict, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@claude / @chatgpt-codex-connector — triage of the latest replies on the green-CI `0892fce` head:

| Comment | Outcome |
|---|---|
| https://github.com/pmatos/s11/pull/172#issuecomment-4476414791 (final-pass approval) | **Already handled (no change)** — verifies items #1 and #2 are resolved; the two "remaining minor" items it notes are tracked as #186 (`count: 16` inline comment) and #187 (`"nzcv,x0"` error UX). |
| https://github.com/pmatos/s11/pull/172#issuecomment-4476435825 (new full review) | Two new suggestions: |
| &nbsp;&nbsp;• `// count must equal 16` inline comment | **Already tracked** as #186 (duplicate of the prior round). |
| &nbsp;&nbsp;• `ParseLiveOutRegistersError` name is "slightly stale" | **Already tracked** as #180 (duplicate of the prior round). |
| &nbsp;&nbsp;• (Optional) Space-separated regs test (`"x0 x1;nzcv"`) | **Deferred (diminishing returns)** — tracked as #188 (https://github.com/pmatos/s11/issues/188). |
| https://github.com/pmatos/s11/pull/172#issuecomment-4476427164 (codex environment stub) | **Already handled** — duplicate of the prior 4476166396 reply; bot infrastructure note. |
| https://github.com/pmatos/s11/pull/172#issuecomment-4476427637 ("I'll analyze and get back") | **Already handled** — Claude bot stub; the analysis that followed (4476435825) is triaged above. |

---

**Status:** All CI green on `0892fce` (test, coverage, clippy, CodeQL, codecov/patch + project, claude-review). All inline review threads resolved (4/4). All actionable PR-level feedback either landed or tracked as follow-up issues #180, #181, #186, #187, #188.

_Triaged by Claude Opus 4.7 and Codex GPT-5.4._
